src/plasmapy/simulation/particle_tracker/particle_tracker.py

```diff
@@ -1014,6 +1016,8 @@ def _update_velocity_stopping(self, summed_field_values) -> None:
             tracked_particles_to_be_stopped_mask
         )
 
+        # Eliminate negative energies before calculating new speeds
+        E = np.where(E < 0, 0, E)
```

Comment:
If E<0, you get an error in the next line in the square root. It doesn't really matter, since these speeds are all set to zero when those particles are set to `stopped` a few lines later, but this avoids the warning.